### PR TITLE
Test coverage increase

### PR DESCRIPTION
**Pull request by cristim**
# Issue Type #

- Feature Pull Request

### CODE REVIEW

File: core/main_test.go [hunk 1]
```diff
@@ -1 +1,92 @@
...
+				drerr: fmt.Errorf("fooErr"),
+			},
+			want:    nil,
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := getRegions(tt.ec2conn)
+			if (err != nil) != tt.wantErr {
```
Comment (xlr-8) [inline core/main_test.go]: Could use https://github.com/cristim/autospotting/blob/master/core/mock_test.go#L11

File: core/main_test.go [hunk 2]
```diff
@@ -1 +1,92 @@
...
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := connectEC2(tt.region)
+			if !reflect.DeepEqual(
+				got.Client.Config.Region,
+				tt.want.Client.Config.Region) {
+				t.Errorf("connectEC2() = %v, want %v",
+					got.Client.Config.Region,
+					tt.want.Client.Config.Region)
```
Comment (xlr-8) [inline core/main_test.go]: Am I understanding this right, or is this not mocked?

**Comment by cristim** [comment]
@xlr-8 please have another look, I addressed your concerns and also did some more things on top.

**Comment by xlr-8** [comment]
LGTM 👍

**Pull request merged**